grammar.js

```diff
@@ -428,8 +427,8 @@ module.exports = grammar({
         "<-",
         field("value", $._expression)
       ),
-    assert: ($) => seq("assert", $._assignment),
-    negation: ($) => seq("!", $._expression_unit),
+    boolean_negation: ($) => seq("!", $._expression_unit),
+    integer_negation: ($) => seq("-", $._expression_unit),
```

Comment:
I'm not sure how this compares to the compiler but this could be refactored into a general `unary_operator` rule which is common in other tree-sitter parsers. It's probably not worth making that refactor if the compiler has BooleanNegation and IntegerNegation AST nodes though or if there aren't many unary operators.

Reply:
As of currently, this is pretty close to how the Gleam parser does it.  There are two types for the different negations (`UntypedExpr::NegateBool` and `UntypedExpr::NegateInt`).  I'll refactor this here when the Gleam parser is refactored :grin:

Reply:
I wouldn't worry too much about matching the Gleam parser. I'm sure you have better ideas than I have